File: go-backend-interview-prep/01_go_basics/01_hello_world/main.py
# Python equivalent of Hello World (compare with main.go)

def main():
    # The simplest Python program.
    # No special entry point required (but we use __main__ convention).
    print("Hello, World!")

    # Print without a trailing newline
    print("Python ", end="")
    print("is ", end="")
    print("interpreted.")

    # Formatted output
    language = "Python"
    year = 1991
    print(f"{language} was released in {year}")


if __name__ == "__main__":
    main()

File: go-backend-interview-prep/01_go_basics/01_hello_world/main.go
package main

import "fmt"

func main() {
	// The simplest Go program.
	// package main + func main() = executable entry point.
	fmt.Println("Hello, World!")

	// Print without a trailing newline
	fmt.Print("Go ")
	fmt.Print("is ")
	fmt.Println("compiled.")

	// Formatted output with Printf
	language := "Go"
	year := 2009
	fmt.Printf("%s was released in %d\n", language, year)
}
